gateway/handler.go

```diff
@@ -352,65 +351,78 @@ func (i *handler) getOrHeadHandler(w http.ResponseWriter, r *http.Request) {
 	}
 	trace.SpanFromContext(r.Context()).SetAttributes(attribute.String("ResponseFormat", responseFormat))
 
-	resolvedPath, contentPath, ok := i.handlePathResolution(w, r, responseFormat, contentPath, logger)
-	if !ok {
+	// TODO: Why did the previous code do path resolution, was that a bug?
+	// TODO: Does If-None-Match apply here?
+	if responseFormat == "application/vnd.ipfs.ipns-record" {
+		logger.Debugw("serving ipns record", "path", contentPath)
+
+		i.addUserHeaders(w)                                 // ok, _now_ write user's headers.
+		w.Header().Set("X-Ipfs-Path", contentPath.String()) // TODO: Should we even set this?
+
+		success := i.serveIpnsRecord(r.Context(), w, r, contentPath, begin, logger)
+
+		if success { // TODO: should we even be reporting this
```

Comment:
yes, this is a global metric for all response types and this is the price to pay for special-casing this response type 
```suggestion
		if success {
```